File: backend/entity/rating.go
package entity

import "gorm.io/gorm"

type Rating struct {
	gorm.Model
	Score    uint    `json:"score"`
	SoundID  uint    `json:"sound_id"`
	Sound    *Sound  `gorm:"foreignKey:SoundID" json:"sound"`
	MemberID uint    `json:"member_id"`
	Member   *Member `gorm:"foreignKey:MemberID" json:"member"`
}

File: frontend/src/interfaces/Rating.ts
import type { Sound } from "./Sound";
import type { Member } from "./Member";

export interface Rating {
  ID: number;
  score: number;
  sound_id: number;
  sound?: Sound;
  member_id: number;
  member?: Member;
}

export interface CreateRatingRequest {
  score: number;
  sound_id: number;
  member_id: number;
}

export interface FindRatingsRequest {
  sound_id?: number;
  member_id?: number;
}
